src/radius_attr.rs

```diff
@@ -114,14 +127,28 @@ fn parse_attribute_content(i:&[u8], t:u8) -> IResult<&[u8],RadiusAttribute> {
         },
         4 => map!(i, take!(4), |v:&[u8]| RadiusAttribute::NasIPAddress(Ipv4Addr::new(v[0],v[1],v[2],v[3]))),
         5 => map!(i, be_u32, |v| RadiusAttribute::NasPort(v)),
-        6 => map!(i, be_u32, |v| RadiusAttribute::ServiceType(v)),
-        7 => map!(i, be_u32, |v| RadiusAttribute::FramedProtocol(v)),
+        6 => map_opt!(i, be_u32, |v| ServiceType::from_u32(v).map(|v| RadiusAttribute::ServiceType(v))),
+        7 => map_opt!(i, be_u32, |v| FramedProtocol::from_u32(v).map(|v| RadiusAttribute::FramedProtocol(v))),
         8 => map!(i, take!(4), |v:&[u8]| RadiusAttribute::FramedIPAddress(Ipv4Addr::new(v[0],v[1],v[2],v[3]))),
         9 => map!(i, take!(4), |v:&[u8]| RadiusAttribute::FramedIPNetmask(Ipv4Addr::new(v[0],v[1],v[2],v[3]))),
-        10 => map!(i, be_u32, |v| RadiusAttribute::FramedRouting(v)),
+        10 => map_opt!(i, be_u32, |v| FramedRouting::from_u32(v).map(|v| RadiusAttribute::FramedRouting(v))),
         11 => value!(i, RadiusAttribute::FilterId(i)),
         12 => map!(i, be_u32, |v| RadiusAttribute::FramedMTU(v)),
-        13 => map!(i, be_u32, |v| RadiusAttribute::FramedCompression(v)),
+        13 => map_opt!(i, be_u32, |v| FramedCompression::from_u32(v).map(|v| RadiusAttribute::FramedCompression(v))),
+        26 => {
+            if i.len() < 5 {
+                return IResult::Incomplete(Needed::Size(5));
+            }
+            value!(
+                i,
+                RadiusAttribute::VendorSpecific(
+                    ((i[0] as u32) << 24) + ((i[1] as u32) << 16) + ((i[2] as u32) << 8) + (i[3] as u32),
+                    &i[4..]
```

Comment:
This would be better written as a subfunction to really parse the VS attribute as described in https://tools.ietf.org/html/rfc2865#section-5.26
I may accept the patch but will need to rewrite this part